pkg/codegen/docs/gen_test.go

```diff
@@ -644,3 +644,227 @@ func TestDecomposeDocstring(t *testing.T) {
 	},
 		info)
 }
+
+func TestGenOverlayResource(t *testing.T) {
+	t.Parallel()
+	dctx := newDocGenContext()
+	testPackageSpec := newTestPackageSpec()
+	testPackageSpec.Resources["prov:module/overlayResource:OverlayResource"] = schema.ResourceSpec{
+		ObjectTypeSpec: schema.ObjectTypeSpec{
+			Description: "This is a module-level resource called OverlayResource.",
+			IsOverlay:   true,
+		},
+		InputProperties: map[string]schema.PropertySpec{
+			"prop": {
+				Description: "An input property.",
+				TypeSpec: schema.TypeSpec{
+					Type: "string",
+				},
+			},
+		},
+	}
+
+	testPackageSpec.Resources["prov:module/overlayResourceConstrainedLanguages:OverlayResourceConstrainedLanguages"] = schema.ResourceSpec{
+		ObjectTypeSpec: schema.ObjectTypeSpec{
+			Description:               "This is a module-level resource called OverlayResourceConstrainedLanguages.",
+			IsOverlay:                 true,
+			OverlaySupportedLanguages: []string{"python", "go", "nodejs"},
+		},
+		InputProperties: map[string]schema.PropertySpec{
+			"prop": {
+				Description: "An input property.",
+				TypeSpec: schema.TypeSpec{
+					Type: "string",
+				},
+			},
+		},
+	}
+
+	testPackageSpec.Resources["prov:module/overlayResourceWrongLanguage:OverlayResourceWrongLanguage"] = schema.ResourceSpec{
+		ObjectTypeSpec: schema.ObjectTypeSpec{
+			Description:               "This is a module-level resource called OverlayResourceWrongLanguage.",
+			IsOverlay:                 true,
+			OverlaySupportedLanguages: []string{"python", "go", "nodejs", "smalltalk"},
+		},
+		InputProperties: map[string]schema.PropertySpec{
+			"prop": {
+				Description: "An input property.",
+				TypeSpec: schema.TypeSpec{
+					Type: "string",
+				},
+			},
+		},
+	}
+
+	schemaPkg, err := schema.ImportSpec(testPackageSpec, nil)
+	assert.NoError(t, err, "importing spec")
+
+	tests := []struct {
+		ResourceName                 string
+		ExpectedLangChooserLanguages string
+	}{
+		{
+			// regular resource, should support all languages (i.e. ExpectedLangChooserLanguages should be empty)
```

Comment:
Nit: we should be consistent on capitalisation of comments. I don't know which we prefer but this is lower and the next is upper.